lib/hooks/queries/useMarketEventHistory.ts

```diff
@@ -0,0 +1,121 @@
+import { useQuery, UseQueryResult } from "@tanstack/react-query";
+import {
+  isIndexedSdk,
+  isRpcSdk,
+  createRpcContext,
+  ZeitgeistIpfs,
+} from "@zeitgeistpm/sdk-next";
+import { getMarket } from "lib/gql/markets";
+import { useSdkv2 } from "../useSdkv2";
+import { Report, MarketDispute } from "@zeitgeistpm/sdk/dist/types";
+
+export const marketsEventsRootQuery = "marketsEvents";
+interface ReportWithTimestamp extends Report {
+  timestamp: number;
+}
+
+interface DisputesWithTimestamp extends MarketDispute {
+  timestamp: number;
+}
+
+export type MarketHistory = {
+  start: {
+    block: number;
+    timestamp: number;
+  };
+  end: {
+    block: number;
+    timestamp: number;
+  };
+  reported: ReportWithTimestamp;
+  disputes: DisputesWithTimestamp[];
+  resolved: string;
+  oracleReported: boolean;
+};
+
+async function getSdk() {
+  return createRpcContext({
+    provider: "wss://zeitgeist.api.onfinality.io/public-ws",
+    storage: ZeitgeistIpfs(),
+  });
+}
+
+export const useMarketEventHistory = (
+  marketId: string,
+): UseQueryResult<MarketHistory> => {
+  const [sdk2, id] = useSdkv2();
+
+  return useQuery(
+    [marketsEventsRootQuery, id, marketId],
+    async () => {
+      if (isIndexedSdk(sdk2) && isRpcSdk(sdk2)) {
+        const sdk = await getSdk();
```

Comment:
You can use the sdk2 here, no need to create a new one

Reply:
There may be an issue with sdk2 then because the same functions for getting the timestamp return different results:

```
const blockHash = await sdk2.api.rpc.chain.getBlockHash(blockNumber);
const timestamp = (await sdk2.api.query.timestamp.now.at(blockHash)).toNumber();
```

sdk2 returns `Jul 25, 2022, 9:46 AM` for `block: 1762884`
sdk returns `Nov 24, 2022, 12:49 PM` for  `block: 1762884`

For market `423` for example, it would appear that what `sdk `returns is correct since `sdk2` returns a date that is before the market creation date.